packages/flutter_tools/lib/src/migrations/swift_package_manager_integration_migration.dart

```diff
@@ -557,69 +565,90 @@ $newContent
             .toList()
             .isNotEmpty;
     if (logErrorIfNotMigrated && !migrated) {
-      logger.printError('PBXNativeTarget was not migrated or was migrated incorrectly.');
+      logger.printError('Some PBXNativeTargets were not migrated or were migrated incorrectly.');
     }
     return migrated;
   }
 
-  List<String> _migrateNativeTarget(List<String> lines, ParsedProjectInfo projectInfo) {
-    if (_isNativeTargetMigrated(projectInfo)) {
-      logger.printTrace('PBXNativeTarget already migrated. Skipping...');
-      return lines;
-    }
-
-    final (int startSectionIndex, int endSectionIndex) = _sectionRange('PBXNativeTarget', lines);
-
-    // Find index where Native Target for the Runner target begins.
-    final ParsedNativeTarget? runnerNativeTarget =
-        projectInfo.nativeTargets
-            .where(
-              (ParsedNativeTarget target) => target.identifier == _runnerNativeTargetIdentifier,
-            )
-            .firstOrNull;
-    if (runnerNativeTarget == null) {
-      throw Exception(
-        'Unable to find parsed PBXNativeTarget for ${_xcodeProject.hostAppProjectName} target.',
-      );
-    }
+  void _migrateNativeApplicationTarget(
+    List<String> lines,
+    ParsedNativeTarget nativeTarget, {
+    required int startSectionIndex,
+    required int endSectionIndex,
+  }) {
+    final String targetName = nativeTarget.name ?? '';
+    final String pbxTargetName =
+        targetName.isNotEmpty ? 'PBXNativeTarget "$targetName"' : 'PBXNativeTarget';
     final String subsectionLineStart =
-        runnerNativeTarget.name != null
-            ? '$_runnerNativeTargetIdentifier /* ${runnerNativeTarget.name} */ = {'
-            : _runnerNativeTargetIdentifier;
-    final int runnerNativeTargetStartIndex = lines.indexWhere(
+        targetName.isNotEmpty
+            ? '${nativeTarget.identifier} /* $targetName */ = {'
+            : nativeTarget.identifier;
+    final int nativeTargetStartIndex = lines.indexWhere(
       (String line) => line.trim().startsWith(subsectionLineStart),
       startSectionIndex,
     );
-    if (runnerNativeTargetStartIndex == -1 || runnerNativeTargetStartIndex > endSectionIndex) {
+
+    if (nativeTargetStartIndex == -1 || nativeTargetStartIndex > endSectionIndex) {
       throw Exception(
-        'Unable to find PBXNativeTarget for ${_xcodeProject.hostAppProjectName} target.',
```

Comment:
Adding the name of the affected target + clarifying that `hostAppProjectName` is the _project_